src/ai/backend/manager/models/base.py

```diff
@@ -1656,3 +1657,30 @@ def generate_sql_info_for_gql_connection(
             "Set 'first' or 'last' to a smaller integer."
         )
     return ret
+
+
+class DecimalType(TypeDecorator, Decimal):
+    """
+    Database type adaptor for Decimal
+    """
+
+    impl = sa.VARCHAR
+    cache_ok = True
+
+    def process_bind_param(
+        self,
+        value: Optional[Decimal],
+        dialect: Dialect,
+    ) -> Optional[str]:
+        return str(value) if value else None
```

Comment:
Like #3272, shouldn't we use an explicit floating format here, or does postgres/sqlalchemy accept scientific notations as well?

Reply:
I fixed it.